modules/dialog/dlg_db_handler.c

```diff
@@ -898,6 +924,77 @@ static inline unsigned int write_pair( char *b, str *name, str *name_suffix,
 	return j;
 }
 
+static int decode_string(str *sin, str *sout)
+{
+	char *p, c, *at;
+
+	int j = 0;
+
+	if (sin == NULL || sout == NULL || sin->s == NULL || sout->s == NULL ||
+		sin->len == 0)
+		return -1;
+	at = sout->s;
+	p = sin->s;
+	while (j != sin->len) {
+
+		if (*p == '\\') {
+			p++;
+			if (*p <= '9') {
+				c = (*p - '0') << 4;
+			} else {
+				c = (*p - 'a' + 10) << 4;
+			}
+			p++;
+			if (*p <= '9') {
+				c = c + (*p - '0');
+			} else {
+				c = c + (*p - 'a' + 10);
+			}
+			*at++ = c;
+			j += 2;
+		} else {
+			*at++ = *p;
+		}
+		p++;
+		j++;
+	}
+	*at = 0;
+	sout->len = at - sout->s;
+
+	return 0;
+}
+
+static int encode_string(str *dst, str *src)
+{
+	unsigned char x;
+	int j = 0;
+	char *at = dst->s;
+	char *p = src->s;
+	while (j <= src->len) {
+		if (*p < 32 || *p > 127 || *p == '#' || *p == '|' || *p == '\\') {
```

Comment:
127 is not printable, so it should be caught by this branch. You could test *p against 127, (*p == 127), since *p can never be greater than 127 (max char).